csrc/host_ir/lower_to_communication.cpp

```diff
@@ -513,70 +517,56 @@ std::vector<Expr*> convertSingleOpToCommunication(
     output_tv->setMemoryType(MemoryType::Global);
   }
 
-  const DeviceMesh& sender_mesh = input_tv->getDeviceMesh();
-  const DeviceMesh& receiver_mesh = output_tv->getDeviceMesh();
-  const bool same_mesh = sender_mesh == receiver_mesh;
-
-  // Stores whether the I/O has its first axis parallelized on DIDx
-  const bool is_input_sharded = isSharded(input_tv) && sender_mesh.size() > 1;
-  const bool is_output_sharded =
-      isSharded(output_tv) && receiver_mesh.size() > 1;
-
   NVF_ERROR(
       isCommunicationLayoutCompliant(e),
       "Resharding on an inner axis is not lowerable ",
       e->toString());
 
-  if (e->isA<LoadStoreOp>()) {
-    if (!is_input_sharded && is_output_sharded) {
-      lowerToScatter(input_tv, output_tv, backend, comms);
-    } else if (is_input_sharded && !is_output_sharded) {
-      if (same_mesh) {
-        lowerToAllgather(input_tv, output_tv, backend, comms, my_device_idx);
-      } else {
-        lowerToGather(input_tv, output_tv, backend, comms);
-      }
-    } else {
-      // TODO(#4604): This is problematic for 2D sharding.
-      lowerToBroadcastOrSendRecv(input_tv, output_tv, backend, comms);
+  CommunicationInfo communication_info = getCommunicationInfo(e);
+
+  BinaryOpType op_type = [](Expr* e) {
+    if (auto* reduce = dynamic_cast<ReductionOp*>(e)) {
+      return reduce->getReductionOpType();
     }
-  } else {
-    BinaryOpType op_type = [&]() {
-      if (auto* reduce = dynamic_cast<ReductionOp*>(e)) {
-        return reduce->getReductionOpType();
-      }
 
-      NVF_ERROR(e != nullptr);
-      if (e->isA<SqueezeOp>()) {
-        return BinaryOpType::Add;
-      }
+    NVF_ERROR(e != nullptr);
+    if (e->isA<SqueezeOp>()) {
+      return BinaryOpType::Add;
+    }
 
-      NVF_THROW("Expected a ReductionOp or a SqueezeOp, but got: ", e);
-    }();
+    NVF_THROW("Expected a ReductionOp or a SqueezeOp, but got: ", e);
```

Comment:
```suggestion
  }(e);
```
**[Suggested by AI]** The closing brace and semicolon were replaced with a closing brace followed by an immediate function-invocation that passes the variable `e` as an argument:  

• Old: `};` simply ended the function definition.  
• New: `}(e);` ends the function body and instantly executes it, supplying `e`.  

This adjustment corrected the build error by properly invoking the self-executing function with the required parameter.